integration-cli/docker_cli_ps_test.go

```diff
@@ -666,3 +666,58 @@ func (s *DockerSuite) TestPsGroupPortRange(c *check.C) {
 	}
 
 }
+
+// Test for GitHub issue #12595
+func (s *DockerSuite) TestPsImageIDAfterUpdate(c *check.C) {
+
+	originalImageName := "busybox:TestPsImageIDAfterUpdate-original"
+	updatedImageName := "busybox:TestPsImageIDAfterUpdate-updated"
```

Comment:
I was attempting to replicate jperville's test case from the issue, as well as Luzifer's original expected/actual report: https://github.com/docker/docker/issues/12595

Their issue looks different to me than what you and didip go on to report later in the thread. I think I need to do some poking around with a client to reconcile the two in my head.